aioredis/cluster/cluster.py

```diff
@@ -0,0 +1,615 @@
+                         commands_factory=commands_factory, loop=loop)
+        self._minsize = minsize
+        self._maxsize = maxsize
+        self._cluster_pool = {}
+
+    def get_nodes_entities(self):
+        return self._cluster_pool.values()
+
+    @asyncio.coroutine
+    def get_cluster_pool(self):
+        cluster_pool = {}
+        nodes = list(self._cluster_manager.masters)
+        tasks = [
+            async_task(create_pool(
+                node.address, db=self._db, password=self._password,
+                encoding=self._encoding, minsize=self._minsize,
+                maxsize=self._maxsize, commands_factory=self._factory,
+                loop=self._loop),
+                loop=self._loop
+            )
+            for node in nodes
+        ]
+        yield from asyncio.gather(*tasks, loop=self._loop)
+
+        for node, task in zip(nodes, tasks):
+            cluster_pool[node.id] = task.result()
+        return cluster_pool
+
+    @asyncio.coroutine
+    def reload_cluster_pool(self):
+        logger.info('Reloading cluster...')
+        yield from self.clear()
+        self._moved_count = 0
+        yield from self.fetch_cluster_info()
+        logger.info('Connecting to cluster...')
+        self._cluster_pool = yield from self.get_cluster_pool()
+        logger.info('Reloaded cluster')
+
+    @asyncio.coroutine
+    def initialize(self):
+        yield from super().initialize()
+        self._cluster_pool = yield from self.get_cluster_pool()
+
+    @asyncio.coroutine
+    def clear(self):
+        """Clear pool connections. Close and remove all free connections."""
+        for pool in self._cluster_pool.values():
+            yield from pool.clear()
+
+    @property
+    def all_slots_covered(self):
+        return self._cluster_manager.all_slots_covered
+
+    def get_pool(self, command, *args, **kwargs):
+        node = self.get_node(command, *args, **kwargs)
+        return self._cluster_pool[node.id]
+
+    @asyncio.coroutine
+    def _execute_node(self, pool, command, *args, **kwargs):
+        """Execute redis command and returns Future waiting for the answer.
+
+        :param command str
+        :param pool obj
+        Raises:
+        * TypeError if any of args can not be encoded as bytes.
+        * ReplyError on redis '-ERR' responses.
+        * ProtocolError when response can not be decoded meaning connection
+          is broken.
+        """
+        cmd = decode(command).lower()
+        try:
+            with (yield from pool) as conn:
+                return (yield from getattr(conn, cmd)(*args, **kwargs))
+        except ReplyError as err:
+            address = parse_moved_response_error(err)
+            if address is None:
+                raise
+            logger.debug('Got MOVED command: {}'.format(err))
+            self._moved_count += 1
+            if self._moved_count >= self.MAX_MOVED_COUNT:
+                yield from self.initialize()
```

Comment:
The same concurrency issue as in parent class, but `minsize * nodes` connections need to be recreated each time.